components/institutional/AsideMenu.tsx

```diff
@@ -1,27 +1,83 @@
+import type { SectionProps } from "$live/mod.ts";
+import Icon from "$store/components/ui/Icon.tsx";
+
 export interface Props {
   menuItems: {
     label: string;
     href: string;
   }[];
 }
 
-function AsideMenu({ menuItems }: Props) {
+export function loader(ctx: Props, req: Request) {
+  const url = new URL(req.url);
+  const { pathname } = url;
+
+  return {
+    ...ctx,
+    pathname,
+  };
+}
+
+function AsideMenu(
+  { menuItems, pathname: currentUrl }: SectionProps<typeof loader>,
+) {
+  const currentRoute = menuItems.find((item) => item.href === currentUrl);
```

Comment:
Inves de mandar a URL para o componente e depois ficar comparando as URLs voce poderia fazer um map do menuItems no loader inline e ja mandar para o componente o objeto pronto. Isso que significa deixar a logica no backend.
```suggestion
export function loader({menuItems}: Props, req: Request) {
  const url = new URL(req.url);
  const { pathname } = url;

  return {
    menuItems: menuItems.map(item =>{...item, isSelected: item.href === pathname})
  };
}

function AsideMenu(
  { menuItems }: SectionProps<typeof loader>,
) {
```